src/gui/src/components/explorer-grid/dependency-sidebar/context.tsx

```diff
@@ -106,6 +128,20 @@ export const DependencySidebarProvider = ({
     })),
   ).current
 
+  /** The `filteredDependencies` by default will the dependencies themselves */
+  useEffect(() => {
+    dependencySidebarStore.setState(() => ({
+      filteredDependencies: dependencies,
+    }))
+  }, [dependencies])
+
+  /** Reset the filters when navigating */
+  useEffect(() => {
+    dependencySidebarStore.setState(() => ({
+      filters: [],
+    }))
+  }, [query])
+
```

Comment:
This seems like it goes against the rule of react that you should not update state based on other state/props inside an effect: https://react.dev/learn/you-might-not-need-an-effect#updating-state-based-on-props-or-state

I'm not exactly sure how it would be refactored without looking closer at how `dependencySidebarStore` is used through the component tree. But in general it seems like we should be able to either calculate the filtered set of dependencies during render or if that is expensive do it inside `useMemo` instead.